lib/features/base/base_controller.dart

```diff
@@ -21,11 +21,13 @@ abstract class BaseController extends GetxController
   FpsCallback? fpsCallback;
 
   void consumeState(Stream<Either<Failure, Success>> newStateStream) async {
-    newStateStream.listen(
-      (state) => onData(state),
-      onError: (error) => onError(error),
-      onDone: () => onDone()
-    );
+    newStateStream
+      .handleError((error) => onError(error))
+      .listen(
+        (state) => onData(state),
+        onError: (error) => onError(error),
```

Comment:
why we need 
```
   .handleError((error) => onError(error))
```
when we have `onError` here?

Reply:
Oh, that's right, it's not necessary

Reply:
please add the log before the `onError`